developer_docs/how_to_create_a_plugin.asciidoc

```diff
@@ -204,25 +203,201 @@ they're easily installable for end users.
+Make sure that you're comfortable with the compatibility
 level of the update, knowing which releases it can safely be run on
 _and_ which it should be updated in. Users on the very old stable
 releases might not expect to receive a new major version of a plugin
 with significant changes, even if it runs OK.
 
+[source,bash]
+----
+cd foreman-packaging
+git checkout rpm/develop
+git log
+commit 8ea39891ccb969e24c7a7de6a0d31c2400a21190 (HEAD -> rpm/develop, origin/rpm/develop)
+Author: Your Name <your_email>
+Date:   Sun Feb 5 04:09:33 2023 +0000
+
+    Add foreman_plugin_template package
+
+...
+git checkout rpm/3.5
+git checkout -b rpm/3.5-foreman_plugin_template-0.0.1
+git cherrypick -x 8ea39891ccb969e24c7a7de6a0d31c2400a21190
+git push origin rpm/3.5-foreman_plugin_template-0.0.1
+----
+
 Lastly, it's helpful for maintainers to open up pull requests for
 packaging updates when making a release to share the workload with the
 regular packaging maintainers. (The regular packagers are also likely to
 be unfamiliar with the plugin and which releases it's appropriate for.)
 
+The workflow is similar to adding it initially.
+
+[source,bash]
+----
+cd foreman_plugin_template
+vi lib/foreman_plugin_template/version.rb
+module ForemanPluginTemplate
+  VERSION = '0.0.2'.freeze
+end
+git commit -m "Release 0.0.2"
+git push origin main
+git tag -a 0.0.2 -m "Release 0.0.2"
+git push origin 0.0.2
+gem build foreman_plugin_template.gemspec
+gem push foreman_plugin_template-0.0.2.gem
+
+cd foreman-packaging
+git add remote upstream https://github.com/theforeman/foreman-packaging.git
+git checkout rpm/develop
+git pull upstream rpm/develop
+git checkout -b rpm/develop-foreman_plugin_template-0.0.2
+./bump_rpm.sh rubygem-foreman_plugin_template
+git push origin rpm/develop-foreman_plugin_template-0.0.2
+----
+
+There is also some automation for this now, trying to keep up with
+releases of plugins and dependencies, but a manual pull request is still
+preferred as plugin developers will know when backports are also needed.
+
+[[puppet-module]]
+=== Puppet module
+
+The next step would be to add your plugin to the puppet module allowing
+for an automated installation. Also fork the repository 
+https://github.com/theforeman/puppet-foreman[puppet-foreman] to your
+account so you can create a pull request. Then you can follow the example.
+
+[source,bash]
+----
+git clone https://github.com/youraccount/puppet-foreman.git
+cd puppet-foreman
+git checkout -b feature/foreman_plugin_template
+vi manifests/plugin/plugin_template.pp
+# = Foreman Plugin Template plugin
+#
+# This class installs plugin_template plugin
+#
+class foreman::plugin::plugin_template {
+  foreman::plugin { 'plugin_template':
+  }
+}
+git add manifests/plugin/plugin_template.pp
+git commit -m "Add foreman plugin for plugin_template"
+----
+
+[[installer-support]]
+=== Installer Support
+
+Last but not least you should add support to the installer for more
```

Comment:
Perhaps make a note that CI will fail until the class is actually available in puppet-foreman's git master branch, but it's fine or even preferred to submit the installer PR even if the puppet-foreman PR is still open.